app/scripts/lib/validate.js

```diff
@@ -28,28 +44,27 @@ define(function (require, exports, module) {
      * @return true if email is valid, false otw.
      */
     isEmailValid: function (email) {
-      if (typeof email !== 'string') {
+      if (typeof email !== 'string' || email.length > 256) {
         return false;
       }
 
       var parts = email.split('@');
+      var localPart = parts[0];
+      var domainPart = parts[1];
+
+      if (parts.length !== 2 || localPart.length > 64) {
+        return false;
+      }
+
+      if (unicodeRegex.test(localPart)) {
+        localPart = punycode.toASCII(localPart);
```

Comment:
With the address converted to punycode, does the backend need to do the same when it sends email? If we allow unicode addresses, do we have to store the converted version locally? If so, does the auth server too? Will endpoints like `/account/status` accept a unicode containing address?